### PR TITLE
masync: add memset operation for synchronous data mover

### PR DESCRIPTION
**Pull request by DamianDuy**
This change is [Reviewable](https://reviewable.io/reviews/pmem/miniasync/72)

### CODE REVIEW

File: src/include/libminiasync/vdm.h [hunk 1]
```diff
@@ -188,6 +202,30 @@ vdm_memmove(struct vdm *vdm, void *dest, void *src, size_t n, uint64_t flags)
 	return future;
 }
 
+/*
+ * vdm_memset -- instantiates a new memset vdm operation and returns a new
+ * future to represent that operation
+ */
+static inline struct vdm_operation_future
```
Comment (pbalcer) [inline src/include/libminiasync/vdm.h:209]: please rebase and use the `vdm_generic_operation` function.

**Comment by kswiecicki** [comment]
___
*[doc/data_mover_sync_get_vdm.3.md, line 49 at r1](https://reviewable.io/reviews/pmem/miniasync/72#-Myr5fsr0gzmAYEBNdof:-Myr5fsr0gzmAYEBNdog:b-y05xaa) ([raw file](https://github.com/pmem/miniasync/blob/097dd52e9f30a75a52f70a68f5889f32142455d6/doc/data_mover_sync_get_vdm.3.md#L49)):*
> ```Markdown
> * **vdm_memcpy**(3) - memory copy operation
> * **vdm_memmove**(3) - memory move operation
> * **vdm_memset**(3) - memory set operation
> ```

This should also be mentioned in the `SEE ALSO` section.



_Code quote:_
```Markdown
**vdm_memset**(3) 
```

**Comment by kswiecicki** [comment]
___
*[doc/miniasync_vdm_synchronous.7.md, line 67 at r1](https://reviewable.io/reviews/pmem/miniasync/72#-Myr6vTDCNKZlPeqKaB-:-Myr6vTM8ZjmuFziU90k:b-dl5760) ([raw file](https://github.com/pmem/miniasync/blob/097dd52e9f30a75a52f70a68f5889f32142455d6/doc/miniasync_vdm_synchronous.7.md#L67)):*
> ```Markdown
> 
> # SEE ALSO #
> ```

Mention `vdm_memset` here too.



_Code quote:_
```Markdown
# SEE ALSO #
```

**Comment by kswiecicki** [comment]
___
*[tests/memset_sync/memset_sync.c, line 56 at r1](https://reviewable.io/reviews/pmem/miniasync/72#-Myr825_DeNfvUPletWZ:-Myr825_DeNfvUPletW_:bczre3g) ([raw file](https://github.com/pmem/miniasync/blob/097dd52e9f30a75a52f70a68f5889f32142455d6/tests/memset_sync/memset_sync.c#L56)):*
> ```C
> 	for (size_t i = 0; i < buffer_size / 2; i++) {
> 		if (buffer[i] != c)
> 			ret = 1;
> ```

You could substitute it with an `UT_ASSERTeq` so that we know what line exactly returned invalid value.



_Code quote:_
```C
ret = 1;
```

**Review by DamianDuy — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/miniasync/72)* status: 0 of 12 files reviewed, 4 unresolved discussions (waiting on @kswiecicki and @pbalcer)

___
*[doc/data_mover_sync_get_vdm.3.md, line 49 at r1](https://reviewable.io/reviews/pmem/miniasync/72#-Myr5fsr0gzmAYEBNdof:-MyvqYvXEP8CJQhTvdtg:b-vl5wcs) ([raw file](https://github.com/pmem/miniasync/blob/097dd52e9f30a75a52f70a68f5889f32142455d6/doc/data_mover_sync_get_vdm.3.md#L49)):*
<details><summary><i>Previously, kswiecicki (Krzysztof Święcicki) wrote…</i></summary><blockquote>

This should also be mentioned in the `SEE ALSO` section.
</blockquote></details>

Done. I've added it there along with `memmove`.
___
*[doc/miniasync_vdm_synchronous.7.md, line 67 at r1](https://reviewable.io/reviews/pmem/miniasync/72#-Myr6vTDCNKZlPeqKaB-:-MyvqmEA6zaX0V__WCGv:b-896fix) ([raw file](https://github.com/pmem/miniasync/blob/097dd52e9f30a75a52f70a68f5889f32142455d6/doc/miniasync_vdm_synchronous.7.md#L67)):*
<details><summary><i>Previously, kswiecicki (Krzysztof Święcicki) wrote…</i></summary><blockquote>

Mention `vdm_memset` here too.
</blockquote></details>

Done.
___
*[src/include/libminiasync/vdm.h, line 209 at r1](https://reviewable.io/reviews/pmem/miniasync/72#-MymoOzMgqeFjm8Ukz3C-r1-209:-MyvqMAVC9w1T0na-L_x:b-896fix) ([raw file](https://github.com/pmem/miniasync/blob/097dd52e9f30a75a52f70a68f5889f32142455d6/src/include/libminiasync/vdm.h#L209)):*
<details><summary><i>Previously, pbalcer (Piotr Balcer) wrote…</i></summary><blockquote>

please rebase and use the `vdm_generic_operation` function.
</blockquote></details>

Done.
___
*[tests/memset_sync/memset_sync.c, line 56 at r1](https://reviewable.io/reviews/pmem/miniasync/72#-Myr825_DeNfvUPletWZ:-MyvqOOI7IBkGgrrUZPi:b-896fix) ([raw file](https://github.com/pmem/miniasync/blob/097dd52e9f30a75a52f70a68f5889f32142455d6/tests/memset_sync/memset_sync.c#L56)):*
<details><summary><i>Previously, kswiecicki (Krzysztof Święcicki) wrote…</i></summary><blockquote>

You could substitute it with an `UT_ASSERTeq` so that we know what line exactly returned invalid value.
</blockquote></details>

Done.

**Review by blazej-smorawski — COMMENTED** [review]
> Reviewed 3 of 12 files at r1, 9 of 9 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/pmem/miniasync/72)* status: all files reviewed, 5 unresolved discussions (waiting on @DamianDuy, @kswiecicki, and @pbalcer)

___
*[doc/vdm_memset.3.md, line 35 at r2](https://reviewable.io/reviews/pmem/miniasync/72#-Mz-2kx_9UZoFEpAu9qd:-Mz-2kx_9UZoFEpAu9qe:b6vlov0) ([raw file](https://github.com/pmem/miniasync/blob/0da02c1270500442dfcc37f128118ad1a9f55c43/doc/vdm_memset.3.md#L35)):*
> ```Markdown
> 	vdm_operation_start op_start;
> 	vdm_operation_check op_check;
> };
> ```

Do we need this inside the vdm_memset manual? I don't think that someone who just wants to perform a memset, needs to see how vdm is build.



_Code quote:_
```Markdown
struct vdm {
	vdm_operation_new op_new;
	vdm_operation_delete op_delete;
	vdm_operation_start op_start;
	vdm_operation_check op_check;
};
```

**Review by pbalcer — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/miniasync/72)* status: all files reviewed, 5 unresolved discussions (waiting on @blazej-smorawski, @kswiecicki, and @pbalcer)

___
*[doc/vdm_memset.3.md, line 35 at r2](https://reviewable.io/reviews/pmem/miniasync/72#-Mz-2kx_9UZoFEpAu9qd:-Mz-74jGEcawAy5QEZX2:bxzfcj) ([raw file](https://github.com/pmem/miniasync/blob/0da02c1270500442dfcc37f128118ad1a9f55c43/doc/vdm_memset.3.md#L35)):*
<details><summary><i>Previously, blazej-smorawski wrote…</i></summary><blockquote>

Do we need this inside the vdm_memset manual? I don't think that someone who just wants to perform a memset, needs to see how vdm is build.
</blockquote></details>

:+1: I removed these from memcpy when I was updating the vdm man page, but memmove and memset were merged immediately after.

**Review by DamianDuy — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/miniasync/72)* status: 8 of 12 files reviewed, 5 unresolved discussions (waiting on @blazej-smorawski, @kswiecicki, and @pbalcer)

___
*[doc/vdm_memset.3.md, line 35 at r2](https://reviewable.io/reviews/pmem/miniasync/72#-Mz-2kx_9UZoFEpAu9qd:-Mz-FYjvE64knCm8hnWI:b-u1wios) ([raw file](https://github.com/pmem/miniasync/blob/0da02c1270500442dfcc37f128118ad1a9f55c43/doc/vdm_memset.3.md#L35)):*
<details><summary><i>Previously, pbalcer (Piotr Balcer) wrote…</i></summary><blockquote>

:+1: I removed these from memcpy when I was updating the vdm man page, but memmove and memset were merged immediately after.
</blockquote></details>

Done. I've deleted it from these man pages. Thanks for noticing.

**Review by pbalcer — APPROVED** [review]
:lgtm:

> Reviewed 3 of 5 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/pmem/miniasync/72#-:-Mzo9yes6j3nP7m73TYf:bnfp4nl)* status: 10 of 12 files reviewed, 4 unresolved discussions (waiting on @blazej-smorawski and @kswiecicki)

**Review by wlemkows — CHANGES_REQUESTED** [review]
> Reviewed 1 of 12 files at r1, 3 of 9 files at r2, 5 of 5 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/pmem/miniasync/72)* status: all files reviewed, 5 unresolved discussions (waiting on @blazej-smorawski, @DamianDuy, and @kswiecicki)

___
*[tests/memset_sync/memset_sync.c, line 64 at r3](https://reviewable.io/reviews/pmem/miniasync/72#-Mzy275o2-P8dHClRzWx:-Mzy275o2-P8dHClRzWy:b-u5hkdn) ([raw file](https://github.com/pmem/miniasync/blob/984f4d2014ed0788f927a9e4b5413441f9168197/tests/memset_sync/memset_sync.c#L64)):*
> ```C
> 	if (ret)
> 		fprintf(stderr,
> 				"incorrect value in buffer after memset");
> ```

It is not true anymore. You have assert in case of incorrect value.

**Review by DamianDuy — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/miniasync/72)* status: 11 of 12 files reviewed, 5 unresolved discussions (waiting on @blazej-smorawski, @kswiecicki, and @wlemkows)

___
*[tests/memset_sync/memset_sync.c, line 64 at r3](https://reviewable.io/reviews/pmem/miniasync/72#-Mzy275o2-P8dHClRzWx:-MzyUf4t42hN2FYtVrPk:bftxz8v) ([raw file](https://github.com/pmem/miniasync/blob/984f4d2014ed0788f927a9e4b5413441f9168197/tests/memset_sync/memset_sync.c#L64)):*
<details><summary><i>Previously, wlemkows (Weronika Lewandowska) wrote…</i></summary><blockquote>

It is not true anymore. You have assert in case of incorrect value.
</blockquote></details>

Done. I've deleted this.

**Review by wlemkows — APPROVED** [review]
> Reviewed 1 of 1 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/pmem/miniasync/72)* status: all files reviewed, 4 unresolved discussions (waiting on @blazej-smorawski and @kswiecicki)

**Pull request merged**